component/resources.jsonnet

```
@@ -15,6 +23,43 @@ local GatewayClass = function(name='') {
   },
 };
 
+local namespacedName(name, namespace='') = {
+  local namespaced = std.splitLimit(name, '/', 1),
+  local ns = if namespace != '' then namespace else params.namespace,
+  namespace: if std.length(namespaced) > 1 then namespaced[0] else ns,
+  name: if std.length(namespaced) > 1 then namespaced[1] else namespaced[0],
+};
+
+local namespaced(obj) = {
+  ['%(namespace)s_%(name)s' % namespacedName(name)]: obj[name] {
+    metadata: {
+      namespace: namespacedName(name).namespace,
+      name: namespacedName(name).name,
+    },
+  }
+  for name in std.objectFields(obj)
```

The way we're using this breaks nulling entries in gateway_parameters. Possible fix;

```suggestion
  [if obj != null then '%(namespace)s_%(name)s' % namespacedName(name)]: obj[name] {
    metadata: {
      namespace: namespacedName(name).namespace,
      name: namespacedName(name).name,
    },
  }
  for name in std.objectFields(obj)
```